platforms/core-configuration/model-core/src/main/java/org/gradle/internal/extensibility/DefaultExtraPropertiesExtension.java

```diff
@@ -79,7 +99,16 @@ public void setProperty(String name, @Nullable Object newValue) {
 
     @Override
     public Map<String, Object> getProperties() {
-        return new HashMap<>(storage);
+        // TODO:configuration-cache use a tracking map here
+        HashMap<String, Object> properties = new HashMap<>(storage);
```

Comment:
💅 

```
Map<String, Object> properties = ...
```

Reply:
I feel slightly inclined to leave the concrete type for the local.